subprojects/testing-base/src/main/java/org/gradle/testing/base/TestSuiteSpecContainer.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.base;
+
+import org.gradle.api.Incubating;
+import org.gradle.model.ModelMap;
+
+/**
+ * A container of {@link TestSuiteSpec} instances.
+ *
+ * This class is actually much older than 8.1, but 8.1 renamed it from TestSuiteContainer,
```

Comment:
Even though this interface was incubating, it existed as far back as 2016. I worry that this change may break existing plugins and buildscripts which already used it with the old name

Reply:
The "old" `TestSuiteContainer` is part of the "software model" types which should not be used too much but there will be scripts and plugins using it in the wild. I agree we should not do the rename as is as it is a breaking change 👍
We need to come up with a name for the new type used for JVM test suites. Using the same name in a different package could work but it naturally fits in `org.gradle.testing.base` which mixes types for the "software model" and the new test suites ... The natural name being already taken the new name will derail from our conventions. What about simply `TestSuites`?

Reply:
I agree these types should be be in light use already. Does it make sense to deprecate them for removal in 9.0 and introduce a migration path? This may be quite involved so perhaps this is a non-option. 

In general, I question why we are even making this change in this PR? It seems unrelated to the `publicType` changes, unless I am missing something. 

Using the same name in a different package is an _option_, though I agree the current package is the correct one, and the current name in this PR would also follow conventions. The two options I see are:
1) Deprecate the existing type, and introduce some migration path. In 9.0 we can use the correct name in the correct package. Though, this would only be possible if we can defer these changes 'till later. 
2) Pick a name like `TestSuites` or `TestSuiteCollection`. 

@big-guy @tresat Thoughts?

Reply:
Deprecating "software model" types is out of scope of this PR. When we do this we should do this as a whole.

We decided to skip the generation of accessors when the extension getter for a container returns a parameterized type, such as `NamedDomainObjectContainer<TestSuite>` because it may lead to conflicting/wrong accessors generated. This change is in this PR to dogfood the introduced feature so that the JVM test suites can benefit from it. This is in line with having a specific type for containers, as we do for e.g. `SourceSetContainer`.

We could defer. This would mean that the DSL for test suites will have to wait until Gradle 10.0 to be nicer with the Kotlin DSL, phew. And we're talking about phasing out software model types for years so it sounds scary to me.

Reply:
> Deprecating "software model" types is out of scope of this PR. When we do this we should do this as a whole.

I was definitely not suggesting this as part of this PR. I understand this may be quite involved. Choosing a non-conventional name might be the best path forward here as this change seems quite nice from a user perspective.

Reply:
> Pick a name like TestSuites 

I'm like this idea, so long as the javadoc on that renamed class explains the reasoning here.